contracts/ExitFormat.sol

```diff
@@ -23,14 +18,35 @@ library ExitFormat {
 
     // A SingleAssetExit specifies
     // * an asset address (0 implies the native asset of the chain: on mainnet, this is ETH)
-    // * custom metadata (optional field, can be zero bytes). This might specify how to transfer this particular asset (e.g. target an "ERC20.transfer"' method)
+    // * custom tokenMetadata
+    //   containing
+    //   - AssetType enum value
+    //   - metadata for that token
     // * an allocations array
     struct SingleAssetExit {
         address asset;
-        bytes metadata;
+        TokenMetadata tokenMetadata;
         Allocation[] allocations;
     }
 
+    // TokenMetadata allows for different token standards
+    // that require additional data than just a token contract address
+    // * assetType specifies one of the supported asset types
+    // * metadata is a differently encoded metadata depending on the token type.
+    //   This is untyped to allow for extensions in future as different token standards emerge
+    struct TokenMetadata {
+        AssetType assetType;
+        bytes metadata;
+    }
+
+    // Enum of different (non-native) token types the SingleAssetExit can contain
```

Comment:
Can we enhance this comment to explain that Null will be used to indicate the native token (perhaps also mentioning that the top level `asset` field is used for that).

Reply:
Expanded this doc comment with a number of other things. Also renamed `Null` to `Native` I think this makes it inherently clearer